dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java

```diff
@@ -117,7 +120,39 @@ public void testPrintStackTraceOverrideReflection() throws Exception {
     }
 
     @Test
-    public void testPrintStackTraceMultipleThreads() {
-        // TODO: Test printStackTrace in multiple threads and ensure that messages are not leaking from one stack trace to another
+    public void testPrintStackTraceMultipleThreads() throws Exception {
```

Comment:
this test needs more than 3 (or 5) threads to do one hit each to check for concurrency problems.  I'd suggest you have 5 threads and make each run in a loop making, say, 1000 hits each... then in your asserts you can check that you got expected number of events, and check each of the 5000 events to make sure they all have the same number of lines in their stack trace.  you could also check to make sure you have exactly 1000 messages from each thread.